### PR TITLE
Additional checks for security concerns during Import YAML and Template process

### PR DESCRIPTION
**Pull request by jwforres**
Also enables the ability to create cluster scoped resources from Import YAML

Implements https://trello.com/c/RCU304BD

### CODE REVIEW

**Comment by jwforres** [comment]
@spadgett think this is ready for a first pass

@liggitt FYI

File: bower.json [hunk 1]
```diff
@@ -47,7 +47,7 @@
     "angular-utf8-base64": "0.0.5",
     "file-saver": "1.3.3",
     "bootstrap-switch": "3.3.3",
-    "origin-web-common": "0.0.3"
+    "origin-web-common": "0.0.5"
```
Comment (jwforres) [inline bower.json]: Yes, won't be able to create cluster resources without this.

Comment (spadgett) [inline bower.json]: Right of course :)

**Comment by jwforres** [comment]
@spadgett ok think this is ready for final review, added a uniq call for roleBindingStrs as well since its only showing the subjects now

**Comment by smarterclayton** [comment]
Here's what I think the whitelist should be:

```
    * buildconfigs (aka 'bc')
    * builds
    * configmaps (aka 'cm')
    * daemonsets (aka 'ds')
    * deployments (aka 'deploy')
    * deploymentconfigs (aka 'dc')
    * endpoints (aka 'ep')
    * events (aka 'ev')
    * horizontalpodautoscalers (aka 'hpa')
    * imagestreamimages (aka 'isimage')
    * imagestreams (aka 'is')
    * imagestreamtags (aka 'istag')
    * ingresses (aka 'ing')
    * jobs
    * persistentvolumeclaims (aka 'pvc')
    * pods (aka 'po')
    * podtemplates
    * replicasets (aka 'rs')
    * replicationcontrollers (aka 'rc')
    * routes
    * secrets
    * serviceaccounts (aka 'sa')
    * services (aka 'svc')
    * statefulsets
```

**Comment by jwforres** [comment]
@spadgett added a second commit with the feedback from @smarterclayton 

can you review those changes, and then i'll squash and rebuild the dist

**Comment by jwforres** [comment]
I put the whitelist in Constants just in case someone yells, but I don't think I want to document it exists

**Comment by spadgett** [comment]
Second commit looks good to me

**Comment by smarterclayton** [comment]
Thank you for your tolerance of my opinions :)

**Comment by jwforres** [comment]
[merge]

**Pull request merged**